### ISSUE TITLE
Possible race condition with graphical console when starting a VM

### ISSUE THREAD

**Issue by hamishmb** [virt-manager/virt-manager#345]
**Distro**: Linux Mint 20.2 Cinnamon x86_64

**Virt-manager version**: 3.0.0

After starting a VM, if I open the graphical console immediately, I get a screen saying the viewer is disconnected, and then if I select serial console 1, and/or switch between consoles, the view starts to work. This only happens when opening the console immediately.

I would test with 3.2.0, but when I download and run from git it still reports the version to be 3.0.0, from the PPA I use, so something seems afoot.

**Comment by crobinso** [comment]
Thanks for the reports, but lets make sure you can get 3.2.0 working correctly first. Make sure all virt-manager instances have been stopped, remove your system installed 3.0.0 version, and do:

```console
git clone https://github.com/virt-manager/virt-manager
cd virt-manager
./virt-manager --debug
```

**Comment by crobinso** [comment]
A lot changed between 3.0.0 and 3.2.0 in this area of the code. We also just fixed a race condition in this area upstream. So if you can reproduce with git, please reopen, but until then I will close this

**Comment by hamishmb** [comment]
I'm pretty sure this is fixed in 3.2.0 after using it a bit on LMDE 5.
